addons/hr/models/hr_plan.py

```diff
@@ -59,6 +82,142 @@ class HrPlan(models.Model):
     _name = 'hr.plan'
     _description = 'plan'
 
+    def _default_trigger_onboarding(self):
+        if self.plan_type == 'onboarding' and self.trigger:
+            return self.trigger
+        return 'manual'
+
+    def _default_trigger_offboarding(self):
+        if self.plan_type == 'offboarding' and self.trigger:
+            return self.trigger
+        return 'manual'
+
     name = fields.Char('Name', required=True)
     plan_activity_type_ids = fields.Many2many('hr.plan.activity.type', string='Activities')
     active = fields.Boolean(default=True)
+    plan_type = fields.Selection(
+        [
+            ('onboarding', 'Onboarding'),
+            ('offboarding', 'Offboarding'),
+            ('other', 'Other'),
+        ],
+        string='Type',
+        default='onboarding',
+        required=True,
+    )
+    trigger_onboarding = fields.Selection(
+        [
+            ('manual', 'Manual'),
+            ('employee_creation', 'Employee Creation'),
+        ],
+        default=_default_trigger_onboarding,
+        required=True,
+        store=False,
+    )
+    trigger_offboarding = fields.Selection(
+        [
+            ('manual', 'Manual'),
+            ('employee_archive', 'Employee Archived'),
+        ],
+        default=_default_trigger_offboarding,
+        required=True,
+        store=False,
+    )
+    trigger_other = fields.Selection(
+        [
+            ('manual', 'Manual'),
+        ],
+        default='manual',
+        required=True,
+        store=False,
+    )
+    trigger = fields.Char(compute='_compute_trigger', store=True)
+    company_id = fields.Many2one(
+         'res.company',
+         string='Company',
+         default=lambda self: self.env.company,
+    )
+
```

Comment:
Could you make it inline if not too long ? I'm not fond of those 10 lines long field definitions.